src/solvers/lowering/byte_operators.cpp

```diff
@@ -180,7 +180,46 @@ static exprt unpack_rec(
       max_bytes,
       ns);
   }
-  else if(ns.follow(src.type()).id()==ID_struct)
+  else if(src.type().id() == ID_complex)
+  {
+    const complex_typet &complex_type = to_complex_type(src.type());
+    const typet &subtype = complex_type.subtype();
+
+    auto subtype_bits = pointer_offset_bits(subtype, ns);
+    CHECK_RETURN(subtype_bits.has_value());
+
+    exprt::operandst byte_operands;
+
+    exprt sub_real = unpack_rec(
```

Comment:
Do you need to check byte alignment? Can I have a complex bitfield for example?

Reply:
I have added `CHECK_RETURN(*subtype_bits % 8 == 0);` above to be safe.